src/main/java/com/contrastsecurity/sdk/scan/CodeArtifactClientImpl.java

```diff
@@ -80,8 +91,28 @@ public CodeArtifactInner upload(final String projectId, final Path file) throws
             + "Content-Transfer-Encoding: binary"
             + CRLF
             + CRLF;
-    final String footer = CRLF + "--" + boundary + "--" + CRLF;
-    final long contentLength = header.length() + Files.size(file) + footer.length();
+    final String metadataSection =
+        metadata != null
+            ? boundaryMarker
+                + CRLF
+                + "Content-Disposition: form-data; name=\"metadata\"; filename=\""
```

Comment:
If we know this form is of type application/json, shouldn't we send it as such instead of sending it as a file? This relates to my question in `CodeArtifactClient` which is "how does the user know how to build this file?"

Reply:
I think this gets back to the API design of the REST endpoint which I did not design.  Let me see if I can provide some clarity on a few of your questions for trying to figure the best way to proceed.

The JSON spec I linked to (https://github.com/Contrast-Security-OSS/contrast-scan-prescan/blob/master/src/main/resources/schema/scan-input-metadata-schema-1.0.0.json) is an ingest specification for the scanner engine itself.  It's saying, that if you want to provide the scanner engine some "prescan data" this is the format it will accept it in.  Its published this way because I just wanted a specification for the input format rather than telling someone, "go read the code", and I also wanted to use code generators like jsonschema2pojo, to automatically generate the libraries to handle the data models of the scanner's prescan ingest format. 

* The scanner makes no promises that this format will **always** be supported, though we will try to keep it backwards compat best as possible, in all future versions of the scanner engine.  The current format represents the simplest format to satisfy the current requirements given the current knowledge of the domain, which we know is not complete.
* The REST API in front of the scanner engine doesn't do anything with this data specifically and merely passes it along to the scanner engine.  From its point of view, it's opaque.
* If the REST api wanted to validate the data before passing to the engine since it knows what engine version it bundles, its free to do that with the publish json schema, but the engine makes no assumptions that will occur.
* For "contrast tools" created to generate prescan data as part of the scan request, they have a well defined json-schema to use to get the right data to the engine. But once again, the engine makes no assumptions that it is guarded behind a REST api or anything thing.

Reply:
> For "contrast tools" created to generate prescan data as part of the scan request

Is this SDK an example of such a "contrast tool", or does the SDK user have to use another tool to generate the file first before they can use the SDK to include in their new code artifacts?